reco_utils/recommender/sar/sar_singlenode.py

```diff
@@ -244,18 +244,21 @@ def fit(self, df):
                 temp_df = self.compute_time_decay(df=temp_df, decay_column=self.col_unity_rating)
             self.unity_user_affinity = self.compute_affinity_matrix(df=temp_df, rating_col=self.col_unity_rating)
 
-        # retain seen items for removal at prediction time
-        self.seen_items = temp_df[[self.col_user_id, self.col_item_id]].values
-
         # affinity matrix
         logger.info("Building user affinity sparse matrix")
         self.user_affinity = self.compute_affinity_matrix(df=temp_df, rating_col=self.col_rating)
 
+        # retain seen items for removal at prediction time
+        seen = temp_df[[self.col_user_id, self.col_item_id]].values
+        self.seen_items = np.ones(self.user_affinity.shape)
+        self.seen_items[seen[:, 0], seen[:, 1]] = -np.inf
+
```

Comment:
what if we remove seen and self.seen_items altogether and leverage the user_affinity since that is already captured, see code change below